Frames grid

NickkB opened the pull request.

Commit:
Add Dialog Box / Image Grid

Commit:
Add frame navigation/frame numbering/loading animation

jcwillox commented on frontend/src/views/LabelView.vue:
```diff
@@ -1,6 +1,11 @@
+    return {
+      dialog: false
+    };
+  }
```
You should use Vue's composition API, the `dialog` reference should be created alongside all the other references in the `<script setup lang="ts">` section

https://vuejs.org/guide/extras/composition-api-faq.html

jcwillox commented on frontend/src/views/LabelView.vue:
```diff
@@ -79,4 +84,85 @@ watchEffect(() => {
+                <v-img
+                  :src="createUrl(framesUrl, frame)"
+                  :aspect-ratio="16 / 9"
+                  cover
```
Try to keep properties grouped together, e.g. group all `:value` props together, additionally properties without a value e.g. `cover` should usually be last.

jcwillox commented on frontend/src/views/LabelView.vue:
```diff
@@ -79,4 +84,85 @@ watchEffect(() => {
+                  style="
+                    border-style: solid;
+                    border-width: 5px;
+                    border-color: yellowgreen;
```
You can simplify this to just `border: 5px solid yellowgreen`. You may also want to pick a colour from the material design color pallet.

https://vuetifyjs.com/en/styles/colors/#material-colors

jcwillox commented on frontend/src/views/LabelView.vue:
```diff
@@ -79,4 +84,85 @@ watchEffect(() => {
+          </v-card-title>
+          <v-row class="d-flex align-center justify-center">
+            <v-col v-for="(frame, i) in frames" :key="i" cols="auto">
+              <div v-if="i === imgIndex">
```
You duplicated this entire section just for 1 CSS style 👀, well you're in for a treat, check out style binding, just use `:style="i === imgIndex && { border: '5px solid yellowgreen' }"` instead and remove this `if` and the duplicated section.

jcwillox commented on frontend/src/views/LabelView.vue:
```diff
@@ -79,4 +84,85 @@ watchEffect(() => {
+                    </v-row>
+                  </template>
+                </v-img>
+                <p class="d-flex align-center justify-center">
```
This can be simplified, drop the `<p>` element and instead add `class="text-center"` to the `v-row` element

jcwillox commented on frontend/src/views/LabelView.vue:
```diff
@@ -79,4 +84,85 @@ watchEffect(() => {
+          <v-card-title>
+            <span class="text-h5"> Select Frame </span>
+          </v-card-title>
+          <v-row class="d-flex align-center justify-center">
```
This can be simplified, to `<v-row justify="center">` you will also need to wrap the element in `<v-card-content>`

jcwillox commented on frontend/src/views/LabelView.vue:
```diff
@@ -79,4 +84,85 @@ watchEffect(() => {
+                  </template>
+                </v-img>
+                <p class="d-flex align-center justify-center">
+                  Frame: {{ i + 1 }}
```
What does this frame number really mean? it's not the actual frame index (e.g. frame 542) and I think its inclusion here is confusing and not necessarily that useful, showing the actual name of the frame which includes the actual frame index, is probably a lot more useful.

jcwillox reviewed (changes requested):
The grid view is looking good so far 👍. I've added my feedback, there are a couple of other changes that need to be made, but best to deal with these ones first, and I'll bring up the other ones later.

Also can you remove all the images you added. When committing in vscode you can select which changes you'd like to commit (by clicking that plus button) and should only include files you intentionally changed.
![image](https://user-images.githubusercontent.com/39120423/168458366-1671e581-19ef-4d69-b6aa-c6f53e221f38.png)

Commit:
Add pull request changes

jcwillox commented on frontend/src/views/LabelView.vue:
```diff
@@ -79,4 +80,55 @@ watchEffect(() => {
+                    "
+                    :src="createUrl(framesUrl, frame)"
+                    :aspect-ratio="16 / 9"
+                    :width="400"
```
Can you drop this size to `250`

jcwillox commented on frontend/src/views/LabelView.vue:
```diff
@@ -79,4 +80,55 @@ watchEffect(() => {
+            Frame: {{ imgIndex + 1 }} of {{ framesList.length }}
+          </v-btn>
+        </template>
+        <v-card>
```
Can you add a toolbar above the `v-card`. To save you time I've already got the toolbar. I ended up creating it as I needed to test the styling, it was kinda glitchy at the moment but got it working properly.
```tsx
<v-toolbar color="primary" class="rounded-t">
  <v-toolbar-title>Select Frame</v-toolbar-title>
  <v-spacer />
  <v-btn icon="mdi-close" @click="dialog = false" />
</v-toolbar>
```

Additionally, to make the dialog expand to the window size, and hide the rounded corners at the top of the card you'll need to add the following and you can also remove the `v-card-title`.

```suggestion
        <v-card style="height: calc(100vh - 104px)" class="rounded-t-0">
```

jcwillox commented on frontend/src/views/LabelView.vue:
```diff
@@ -79,4 +80,55 @@ watchEffect(() => {
+                      imgIndex = i;
+                      dialog = false;
+                    "
+                    cover
```
Can you remove `cover` otherwise we end up clipping 4/3 images

jcwillox reviewed (changes requested):
Awesome, just a few small changes left.

Commit:
Add pull request changes

- Add tool bar to dialog box
- Remove original tool bar title
- Remove cover prop on images
- Change image width to 250

jcwillox committed:
Merge branch 'main' into FramesGrid
